### PR TITLE
Feature/tab bar manage active state

### PR DESCRIPTION
**Pull request by Sarin-Udompanish**
## Description
From accessibility spec of tab-bar when user press tab button it should focus on current active tab and when press tab again it should focus out of this tab bar. That mean user can press tab button only once per tab. 

To achieve this we need to set tabIndex of other tab to -1 and when the active tab changed it should reassign all tabIndex. The problem is currently our active state of tab-bar is managed by user so user need to reassign tab index by themselves and that not what we want.

## Type of change
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Checklist
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: packages/elements/src/tab-bar/index.ts [hunk 1]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
   }
 
   /**
-   * Hide all scroll buttons
-   * @returns {void}
+   * Return true if tab list have matched value
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:157]:
```suggestion
   * Return true if incoming value matches one of the existing tab
```

File: packages/elements/src/tab-bar/index.ts [hunk 2]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
-    if (scrollLeft > 0) {
-      this.leftBtn.style.setProperty('display', 'flex');
+    if (activeTab) {
+      this.value = this.getTabValue(activeTab);
     }
-    else {
-      this.leftBtn.style.setProperty('display', 'none');
+    // wait element size updated
+    setTimeout(() => {
+      this.toggleScrollButton(this.content.clientWidth);
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:183]: Ideally, I think this should be handled when element resizes

File: packages/elements/src/tab-bar/index.ts [hunk 3]
```diff
@@ -11,15 +11,16 @@ import { customElement } from '@refinitiv-ui/core/lib/decorators/custom-element.
 import { property } from '@refinitiv-ui/core/lib/decorators/property.js';
 import { query } from '@refinitiv-ui/core/lib/decorators/query.js';
 import { VERSION } from '../version.js';
-import type { Tab } from '../tab';
 import { tweenAnimate } from './helpers/animate.js';
+import { Tab } from '../tab';
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:15]: Please add .js extension.

File: packages/elements/src/tab-bar/index.ts [hunk 4]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
+      this.toggleScrollButton(this.content.clientWidth);
+    });
+  }
+
+  /**
+   * Mark tab as active
+   * @param value value of tab to select
+   * @returns {void}
+   */
+  private selectValue (value: string): void {
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:192]: I think using `active` across the code would be easier to navigate, using `select` may cause confusion with the element state. Something like `activateTab`, `setActiveTab` etc. would sound better.

File: packages/elements/src/tab-bar/index.ts [hunk 5]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
+   */
+  private getTabLabel (tab: Tab): string {
+    return tab.label || tab.textContent || '';
+  }
+
+  /**
+   * Get Tab elements from slot
+   * @returns the array of Tab
+   */
+  private getTabElements () {
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:252]: Please add return type.

File: packages/elements/src/tab-bar/index.ts [hunk 6]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
    * @returns {void}
    */
-  private toggleScrollButton (elementWidth: number): void {
-    const { scrollLeft, scrollWidth } = this.content;
-
-    if (this.vertical) {
+  private onSlotChange (): void {
+    this.setLevel();
+    const tabList = this.getFocusableTabs();
+    if (tabList.length < 1) {
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:173]: Can we check `tabList` length before setting the level ?

File: packages/elements/src/tab-bar/index.ts [hunk 7]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
+   * Hide/Show scroll button when element is overflow.
+   * @param elementWidth width of element
    * @returns {void}
    */
-  private onSlotChange (): void {
-    this.setLevel();
+  private toggleScrollButton (elementWidth: number): void {
+    const { scrollLeft, scrollWidth } = this.content;
+
+    if (this.vertical) {
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:289]: Can this be the first statement of the function ?

File: packages/elements/src/tab/index.ts [hunk 8]
```diff
@@ -88,26 +104,17 @@ export class Tab extends ControlElement {
...
-  /**
-   * @param node that should be checked
-   * @returns whether node can be ignored.
-   */
-  private isIgnorable (node: Node): boolean {
-    return node.nodeType === document.COMMENT_NODE
-      || isAllWhitespaceTextNode(node);
-  }
+  @state()
+  private isSlotHasContent = true;
```
Comment (wsuwt) [inline packages/elements/src/tab/index.ts:108]: Why is this `true` ?

File: packages/elements/src/tab/index.ts [hunk 9]
```diff
@@ -88,26 +104,17 @@ export class Tab extends ControlElement {
...
 
   /**
-   * Checks slotted children nodes and updates component to refresh label and sub-label templates.
+   * Run on default slot slotchange
```
Comment (wsuwt) [inline packages/elements/src/tab/index.ts:111]:
```suggestion
   * Triggers when slot detect changes
```

File: packages/elements/src/tab-bar/index.ts [hunk 10]
```diff
@@ -134,61 +154,147 @@ export class TabBar extends ResponsiveElement {
...
+        tab.active = false;
+      }
+    });
+  }
 
-    // handle right button
-    if (Math.floor(scrollWidth - scrollLeft) > Math.round(elementWidth)) {
-      this.rightBtn.style.setProperty('display', 'flex');
+  /**
+   * If the target of the event is a Tab, then set the value of the target to the value of the Tab
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:212]:
```suggestion
   * Set tab value and fires `tab-changed` event
```

File: packages/elements/src/tab-bar/index.ts (hunk 2 above)
Comment (Sarin-Udompanish) [reply packages/elements/src/tab-bar/index.ts]: fixed

File: packages/elements/src/tab-bar/index.ts [hunk 11]
```diff
@@ -134,61 +155,145 @@ export class TabBar extends ResponsiveElement {
...
-   * Hide all scroll buttons
-   * @returns {void}
+   * Return true if incoming value matches one of the existing tabs
+   * @param value Value to check
+   * @returns true if incoming value matches one of the existing tabs
    */
-  private hideScrollButtons (): void {
-    this.leftBtn.style.setProperty('display', 'none');
-    this.rightBtn.style.setProperty('display', 'none');
+  private isValidValue (value: string) {
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:162]:
```suggestion
  private isValidValue (value: string): boolean {
```

File: packages/elements/src/tab-bar/index.ts [hunk 12]
```diff
@@ -134,61 +155,145 @@ export class TabBar extends ResponsiveElement {
...
+  private toggleScrollButton (elementWidth: number): void {
+    if (this.vertical) {
+      return;
+    }
+
+    const { scrollLeft, scrollWidth } = this.content;
+
+    const leftBtnStyle = scrollLeft > 0 ? 'flex' : 'none';
+    const rightBtnStyle = scrollWidth - scrollLeft - elementWidth > 1 ? 'flex' : 'none';
+
```
Comment (wsuwt) [inline packages/elements/src/tab-bar/index.ts:293]:
```suggestion
```

**Review by wsuwt — APPROVED** [review]

**Pull request merged**